scylla/src/statement/mod.rs

```diff
@@ -24,6 +25,8 @@ pub struct StatementConfig {
     pub request_timeout: Option<Duration>,
 
     pub history_listener: Option<Arc<dyn HistoryListener>>,
+
+    pub execution_profile_handle: Option<ExecutionProfileHandle>,
```

Comment:
I thought that with the addition of the execution profiles, we would remove the possibility of setting individual policies in the `StatementConfig`. Leaving it here causes us to consider 3 sources of policies for each query: the default one, the one in execution profile & one in `StatementConfig`. I feel like that's a lot.

Also, because of it, we are cloning `Arc`/`Box` of retry,speculative,load balancing policies on each `_iter` call to provide those policies to the iterator. With the introduction of execution profiles (and getting rid of those policies in `StatementConfig`) we could replace all this cloning with a single `ExecutionProfileHandle` clone.

For reference - [Java driver does not permit setting individual policies in `Statement`](https://docs.datastax.com/en/drivers/java/4.10/com/datastax/oss/driver/api/core/cql/Statement.html#setQueryTimestamp-long-)

Reply:
I think alike. That is, with your suggestion we get both simpler design and less cloning. WDYT @piodul?

Reply:
> I thought that with the addition of execution profiles, we would remove the possibility of setting individual policies in the `StatementConfig`. Leaving it here causes us to consider 3 sources of policies for each query: the default one, the one in execution profile & one in `StatementConfig`. I feel like that's a lot.
> 
> Also, because of it, we are cloning `Arc`/`Box` of retry,speculative,load balancing policies on each `_iter` call to provide those policies to the iterator. With the introduction of execution profiles (and getting rid of those policies in `StatementConfig`) we could replace all this cloning with a single `ExecutionProfileHandle` clone.
> 
> For reference - [Java driver does not permit setting individual policies in `Statement`](https://docs.datastax.com/en/drivers/java/4.10/com/datastax/oss/driver/api/core/cql/Statement.html#setQueryTimestamp-long-)

Makes sense. Judging by the design of the Java driver's API, setting policies on a per-statement basis is not common, and - if somebody really wishes to do it - they can still do it in a less ergonomic way, i.e. create a new profile derived/cloned from the current one and only change the policy/policies.

Just a nitpick: we should rather clone `ExecutionProfile` than `ExecutionProfileHandle` when executing a query. We want to use the same configuration for all speculative executions/retries/pages in case of `_iter` methods etc.